### ISSUE TITLE
for a VM /kernel should be optional

### ISSUE THREAD

**Issue by LeeSmet** [threefoldtech/zos#1624]
As follow up of #1622, the `/kernel` file can be made optional. The default can be a firmware blob which simply boots the kernel in the boot partition (e.g. https://github.com/cloud-hypervisor/rust-hypervisor-firmware)

**Comment by muhamadazmy** [comment]
Cloud init requires a 'config source' this can be a service which means the node must have network configured already before it can fetch config. This of course does not make sense on the grid because we need cloud-init initially for setting up the network since we don't use dhcp.

Another option is to provide a new data source for cloud-init that uses the kernel params to build proper cloud-init config. But this means that image builders then need to make sure to include this plugin in their image.

Another way to work around this is that we only support official images that we provide in that case we don't allow people to run 'any' image on the chain but rather select one of official images (ubuntu, fedor, centos, etc...) in that case we can either include our custom data source or initramfs scripts.

The point is i don't want to again separate how we handle VMs against Containers. The point of the current approach is to unify the network and storage setup for both use cases. 

Last (but least flexible) is that we always use our own kernel+initramfs that knows exactly how to handle kernel params to setup the VM. In that case the user need to provide only the base image.

So to sum up, we have the following options:
- Custom cloud-init source that can interpret kernel params and provide proper cloud-init config
- Use always our own kernel 👎🏼 user only provide base image
   - We need to figure out how to auto resize the base image to fit into the allocated virtual disk (this is why we require btrfs since zos can only deal with this fs type)
- We provide VM mode only for official images that include either the custom cloud-init source or init scripts.

**Comment by muhamadazmy** [comment]
I checked cloud-init modules and there is a 'grow-part' modules that (I think) we can use to grow the disk image to take the full allocated disk which if done properly can drop the requirements of a btrfs filesystem for disk images

**Comment by LeeSmet** [comment]
Actually for now we can use the `nocloud` datasource https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html. This can inject the config through a CDROM or a file, and then extract the networking from that. This seems like it would allow unmodified cloud init images. 

The user does not NEED to have a cloud init enabled image either. If we document how the config is set, he can also include custom code to handle that if that is what he prefers.
